kafka-client/src/test/java/dev/responsive/kafka/internal/db/rs3/client/grpc/GrpcRS3ClientTest.java

```diff
@@ -148,6 +158,69 @@ public void shouldGetOffsets() {
     assertThat(offsets.flushedOffset(), is(Optional.of(3L)));
   }
 
+  @Test
+  public void shouldRetryGetOffsets() {
+    // given:
+    when(stub.getOffsets(any()))
+        .thenThrow(new StatusRuntimeException(Status.UNAVAILABLE))
+        .thenThrow(new StatusRuntimeException(Status.UNAVAILABLE))
+        .thenReturn(
+            Rs3.GetOffsetsResult.newBuilder()
+                .setWrittenOffset(13)
+                .setFlushedOffset(3)
+                .build());
+
+    // when:
+    final var offsets = client.getCurrentOffsets(STORE_ID, LSS_ID, PSS_ID);
+
+    // then:
+    assertThat(offsets.writtenOffset(), is(Optional.of(13L)));
+    assertThat(offsets.flushedOffset(), is(Optional.of(3L)));
+  }
+
+  @Test
+  public void shouldPropagateUnexpectedExceptionFromGetOffsets() {
+    // given:
+    when(stub.getOffsets(any()))
+        .thenThrow(new StatusRuntimeException(Status.UNKNOWN));
+
+    // when:
+    final RS3Exception exception =
+        assertThrows(RS3Exception.class, () -> client.getCurrentOffsets(STORE_ID, LSS_ID, PSS_ID));
+
+    // then:
+    assertThat(exception.getCause(), is(instanceOf(StatusRuntimeException.class)));
+    assertThat(((StatusRuntimeException) exception.getCause()).getStatus(), is(Status.UNKNOWN));
+  }
+
+  @Test
+  public void shouldTimeoutGetOffsets() throws Exception {
```

Comment:
Can we implement this test by advancing time in `MockTime` on every call to `getOffsets`. That should make it faster.

Reply:
Yeah, that's a nice suggestion.

Reply:
In fact, we don't need to sleep at all from the test. The backoff will advance time for us.